### PR TITLE
Added ability to retrieve notes

### PR DESCRIPTION
**Pull request by Nigel2x4**
Added ability to retrieve a post's note information.

### CODE REVIEW

File: pytumblr/__init__.py [hunk 1]
```diff
@@ -488,6 +488,23 @@ def edit_post(self, blogname, **kwargs):
         valid_options = ['id'] + self._post_valid_options(kwargs.get('type', None))
         return self.send_api_request('post', url, kwargs, valid_options)
 
+    @validate_blogname
+    def notes_post(self, blogname, id, **kwargs):
```
Comment (komapa) [inline pytumblr/__init__.py]: I think the name should be either just `notes` or `post_notes`. We can only have notes on posts so the first name would be preferred. Thoughts?

Comment (Nigel2x4) [reply pytumblr/__init__.py]: That makes sense, can't remember why I had picked notes_post for the name before.  I have included a commit to change the name to notes.

File: pytumblr/__init__.py [hunk 2]
```diff
@@ -488,6 +488,23 @@ def edit_post(self, blogname, **kwargs):
...
+
+        :param blogname: a string, the url of the blog that houses the post
+        :param id: a string, the id of the post.
+        :param mode: a string. Undocumented. Automatically added by tumblr but it's use is not yet known.
+        :param before_timestamp: a string, retreives data before this timestamp
+
+        :returns: a dict created from the JSON response
+        """
+        url = "/v2/blog/{}/notes".format(blogname)
+        valid_options = ['id', "mode", "before_timestamp"]
```
Comment (komapa) [inline pytumblr/__init__.py]: Nitpick but can we stick to single or double quotes for all of the elements here? Thank you :)

**Comment by komapa** [comment]
Thank you for the rename. Can we add also a quick unit test for the new method and we should be ready to merge here!

**Comment by Nigel2x4** [comment]
Added a test and fixed the quotations. Should be good now! Thank you for your patience.

**Comment by komapa** [comment]
This looks great. I just want to make sure you did follow the step in the CONTRIBUTING guide before we can merge the PR: https://github.com/tumblr/pytumblr/blob/master/CONTRIBUTING.md
Once that is done, we can get it in and release a new version.

**Comment by komapa** [comment]
@Nigel2x4 bump for signing the CLA. The CLA, contact information, and GitHub sign-in can be found here: https://yahoocla.herokuapp.com.

**Comment by Nigel2x4** [comment]
I should've already had that, did it reject me?

**Pull request merged**